scripts/genplugins.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"encoding/json"
+	"go/ast"
+	"go/parser" // provides methods for parsing source files and generating asts
+	"go/token"  // provides types and methods for Go's lexer processor (tokenization)
+	"io/ioutil"
+	"strings"
+)
+
+func main() {
+	fset := token.NewFileSet()
+
+	file, err := parser.ParseFile(fset, "../helper/builtinplugins/registry.go", nil, parser.ParseComments)
+	if err != nil {
+		panic(err)
+	}
+
+	cmap := ast.NewCommentMap(fset, file, file.Comments)
+
+	v := Visitor{cmap, make([]Backend, 0), make([]Backend, 0), make([]Backend, 0)}
+
+	ast.Walk(&v, file)
+
+	output, _ := json.MarshalIndent(v, "", " ")
+	_ = ioutil.WriteFile("plugins.json", output, 0644)
+}
+
+type Backend struct {
+	Name       string `json:"name"`
+	Deprecated bool   `json:"deprecated"`
+}
+
+type Visitor struct {
+	// Comment map
+	commentMap ast.CommentMap
+
+	// Credential backends
+	CredBackends []Backend `json:"CredentialBackends"`
+
+	// Logical backends
+	LogBackends []Backend `json:"LogicalBackends"`
+
+	// DB backends
+	DbBackends []Backend `json:"DbPlugins"`
+}
+
+func (v *Visitor) Visit(n ast.Node) ast.Visitor {
+	if n == nil {
+		return nil
+	}
+
+	if keyValueExpr, ok := n.(*ast.KeyValueExpr); ok {
+		if ident, ok := keyValueExpr.Key.(*ast.Ident); ok {
```

Comment:
Style: it might be better to invert the `if` logic here to flatten out the code structure:

```go
keyValueExpr, ok := n.(*ast.KeyValueExpr)
if !ok {
     return v
}

ident, ok := keyValueExpr.Key.(*ast.Ident)
if !ok {
     return v
}

// the rest of the code
```